contracts/InterestRate/StableRateModel.sol

```diff
@@ -0,0 +1,135 @@
+     * @param stableRatePremium_ The multiplierPerBlock after hitting a specified utilization point
+     * @param optimalStableLoanRatio_ Optimal stable loan rate percentage.
+     * @param owner_ Address of the owner for this model(Governance)
+     */
+    constructor(
+        uint256 baseRatePerYear_,
+        uint256 stableRatePremium_,
+        uint256 optimalStableLoanRatio_,
+        address owner_
+    ) public {
+        owner = owner_;
+
+        updateStableRateModelInternal(baseRatePerYear_, stableRatePremium_, optimalStableLoanRatio_);
+    }
+
+    /**
+     * @notice Updates the parameters of the interest rate model (only callable by owner, i.e. Timelock)
+     * @param baseRatePerYear_ The approximate target base APR, as a mantissa (scaled by BASE)
+     * @param stableRatePremium_ The multiplierPerBlock after hitting a specified utilization point
+     * @param optimalStableLoanRatio_ Optimal stable loan rate percentage.
+     * @notice Emits NewStableInterestParams, after updating the parameters
+     * @notice Only governance
+     */
+    function updateStableRateModel(
+        uint256 baseRatePerYear_,
+        uint256 stableRatePremium_,
+        uint256 optimalStableLoanRatio_
+    ) external {
+        require(msg.sender == owner, "StableRateModel: only owner may call this function.");
+
+        updateStableRateModelInternal(baseRatePerYear_, stableRatePremium_, optimalStableLoanRatio_);
+    }
+
+    /**
+     * @notice Calculates the ratio of the stable borrows to total borrows
+     * @param stableBorrows The amount of stable borrows in the market
+     * @param totalBorrows The amount of total borrows in the market
+     * @return The stable loan rate as a mantissa between [0, BASE]
+     */
+    function stableLoanRatio(uint256 stableBorrows, uint256 totalBorrows) public pure returns (uint256) {
+        // Loan ratio is 0 when there are no stable borrows
+        if (totalBorrows == 0) {
+            return 0;
+        }
+
+        return (stableBorrows * BASE) / totalBorrows;
+    }
+
+    /**
+     * @notice Calculates the current borrow rate per block, with the error code expected by the market
+     * @param stableBorrows The amount of stable borrows in the market
+     * @param totalBorrows The amount of borrows in the market
+     * @param variableBorrowRate Current variable borrow rate per block of the protocol
+     * @return The borrow rate percentage per block as a mantissa (scaled by BASE)
+     */
+    function getBorrowRate(
+        uint256 stableBorrows,
+        uint256 totalBorrows,
+        uint256 variableBorrowRate
+    ) external view returns (uint256) {
+        uint256 loanRatio = stableLoanRatio(stableBorrows, totalBorrows);
+        uint256 excessLoanRatio = calculateLoanRatioDiff(loanRatio);
+
+        return (variableBorrowRate + baseRatePerBlock + ((stableRatePremium * excessLoanRatio) / BASE));
```

Comment:
Here also we should use SafeMath

Reply:
Fixed in [commit](https://github.com/VenusProtocol/venus-protocol/pull/210/commits/482b814841c2e68904e01250a89f48d5c745e83a)